includes/collection/class-inbox.php

```diff
@@ -140,4 +141,93 @@ public static function get( $guid, $user_id ) {
 
 		return \get_post( $post_id );
 	}
+
+	/**
+	 * Get an inbox item by its GUID.
+	 *
+	 * @param string $guid The GUID of the inbox item.
+	 *
+	 * @return \WP_Post|\WP_Error The inbox item or WP_Error.
+	 */
+	public static function get_by_guid( $guid ) {
+		global $wpdb;
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		$post_id = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT ID FROM $wpdb->posts WHERE guid=%s AND post_type=%s",
+				\esc_url( $guid ),
+				self::POST_TYPE
+			)
+		);
+
+		if ( ! $post_id ) {
+			return new \WP_Error(
+				'activitypub_inbox_item_not_found',
+				\__( 'Inbox item not found', 'activitypub' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		return \get_post( $post_id );
+	}
+
+	/**
+	 * Undo a received activity.
+	 *
+	 * @param string $id The ID of the inbox item to be removed.
+	 *
+	 * @return bool|\WP_Error True on success, WP_Error on failure.
+	 */
+	public static function undo( $id ) {
+		$type = '';
+		$post = self::get_by_guid( $id );
```

Comment:
```suggestion
		$inbox_item = self::get_by_guid( $id );
```

Could this be `$inbox_item` or something along the lines?

I was about to suggest using `$post->post_author` directly in line 202 but realized that `$post` was being overwritten by the `Remote_Actors::get_by_uri( $actor )` call in line 196. Maybe that `$post` could be called `$remote_actor` or `$actor_post`? Something to make it less ambiguous.